File: BaekJoon/solve/1/2309/Permutation.cpp
#include <iostream>
#include <vector>

using namespace std;

int nums[5] = {1, 2, 3, 4, 5};
bool visited[5];

void make_permutation(vector<int> current_nums) {
  if(current_nums.size() == 3) {
    for(int n : current_nums) { 
      cout << n << " ";
    }
    cout << "\n";
    return;
  }

  for(int i = 0; i < 5; ++i) { 
    if(visited[i] == true) { 
      continue;
    } else { 
      visited[i] = true;
      vector<int> next_nums = current_nums;
      next_nums.push_back(nums[i]);
      make_permutation(next_nums);
      visited[i] = false; 
    }
  }
}

int main(void) { 
  make_permutation({ });
  return 0;
}

File: BaekJoon/solve/1/2309/Permutation.swift
let nums = [1, 2, 3, 4, 5, 6]
var visited = Array(repeating: false, count: nums.count)

func makePermutation(currentNums: [Int]) { 
  if currentNums.count == 3 { 
    print(currentNums)
  }

  for i in 0..<nums.count { 
    if visited[i] == true { 
      continue
    } else { 
      visited[i] = true
      makePermutation(currentNums: currentNums + [nums[i]])
      visited[i] = false
    }
  }
}

makePermutation(currentNums: [])